Sentry: Issue details and assign action

thomaspaulmann opened the pull request:
## Description

I adopted the new `@raycast/utils` to the Sentry extension. Furthermore I added a detail view and an action to assign issues.

## Screencast

![sentry 2022-08-19 at 18 04 53](https://user-images.githubusercontent.com/12066405/185671047-be0a93c8-1662-45db-8020-ee25cc5a7e22.png)

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

mattisssa commented:
Hm, I got into this screen:

![CleanShot 2022-08-21 at 12 29 12@2x](https://user-images.githubusercontent.com/15333662/185784734-73283032-9231-40c4-922d-4774c75908e3.png)

```
Error: Worker terminated due to reaching memory limit: JS heap out of memory

__node_internal_captureLargerStackTrace:errors:464:5
NodeError:errors:371:5
Worker.[kOnExit]:worker:279:26
Worker.<computed>.onexit:worker:201:20
12:29:16.636 
Error: Worker terminated due to reaching memory limit: JS heap out of memory

__node_internal_captureLargerStackTrace:errors:464:5
NodeError:errors:371:5
Worker.[kOnExit]:worker:279:26
Worker.<computed>.onexit:worker:201:20
```

The Store version works fine tho.

sxn commented on extensions/sentry/src/sentry.ts:
```diff
@@ -0,0 +1,57 @@
+    searchParams.append("project", projectId);
+  }
+
+  return useFetch<User[]>(`https://sentry.io/api/0/organizations/${organizationSlug}/users/?` + searchParams, {
```
i -think- this introduces a new scope requirement. My old sentry API key, which has access to the scopes mentioned in the README (`event:read` and `project:read`) doesn't seem to work, or at least not fully:
<img width="645" alt="image" src="https://user-images.githubusercontent.com/1155589/186885330-9f6cd09d-162a-4210-a8b4-ab1072037699.png">

I generated an API key that has access to `org:read`, too, and that one seems to work as it should.

thomaspaulmann replied:
I explained the new scopes in the README and also added a proper view for the unauthorized state. I hope this is enough for a "smooth" transition.

sxn commented:
not a must, but a custom [EmptyView](https://developers.raycast.com/api-reference/user-interface/list?q=emptyview#list.emptyview) could be a nice touch 😺

<img width="812" alt="image" src="https://user-images.githubusercontent.com/1155589/186885951-74fc9f49-babd-46a7-949f-8854255bf76e.png">

thomaspaulmann commented:
> not a must, but a custom [EmptyView](https://developers.raycast.com/api-reference/user-interface/list?q=emptyview#list.emptyview) could be a nice touch 😺
> 
> <img alt="image" width="812" src="https://user-images.githubusercontent.com/1155589/186885951-74fc9f49-babd-46a7-949f-8854255bf76e.png">

I love custom empty views!

pernielsentikaer reviewed (changes requested):
Hi @thomaspaulmann 👋 

If using the wrong `Auth Token` I just get this view

![CleanShot 2022-09-12 at 14 43 29@2x](https://user-images.githubusercontent.com/38033107/189656864-972109f0-e3dd-4a8f-9168-207752ce0efb.png)

It would be nice to get the loader to stop and show the `List.Emptyview` instead

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

thomaspaulmann commented:
<img width="1000" alt="sentry 2022-09-14 at 11 28 56" src="https://user-images.githubusercontent.com/12066405/190130738-3066edcf-be37-45b0-b661-0be776983e53.png">

@pernielsentikaer I added a nice error screen and also updated the README with the new required scopes.

pernielsentikaer commented:
@thomaspaulmann did you push the changes yet?

thomaspaulmann commented:
@pernielsentikaer something was off with my local setup. Pushed my changes!

pernielsentikaer reviewed (approved):
Looks good to me, approved 🔥

thomaspaulmann commented:
The [docs](https://docs.sentry.io/api/auth/) state that this should also work with self-hosted instaces.

NoorBayari commented:
Hi @thomaspaulmann, first of all, thank you for creating this extension - it's super useful! I just wanted to clarify that although tokens can be created on a self-hosted Sentry instance, they can't be used on sentry.io. One potential solution could be to add a feature that allows configuring the URL, so the extension can communicate with a self-hosted instance instead. Hope that helps!

thomaspaulmann commented:
That should be possible. Can you send me some example cURL request that works for you so that I can make the changes. Unfortunately I don’t have access to a self-hosted instance of Sentry to test it.
